.travis.yml

```diff
@@ -1,53 +1,50 @@
-    # The SDK version used for the emulator
-    - android-$EMULATOR_API_VERSION
-
-    #system images
-    - sys-img-armeabi-v7a-android-$EMULATOR_API_VERSION
+    - BUILD_TOOLS_VERSION=29.0.3
+    - API_VERSION=29
+    - ANDROID_TOOLS=commandlinetools-linux-6609375_latest.zip
+    - ARCH=x86_64
 
 before_install:
+  # download and unzip Android SDK command line tools
+  - sudo apt install openjdk-8-jdk
+  - export JAVA_HOME=/usr/lib/jvm/java-8-openjdk-amd64 
+  - echo $JAVA_HOME
+  #- sudo update-alternatives --set java /usr/lib/jvm/java-8-openjdk-amd64/jre/bin/java
+  - wget -nv https://dl.google.com/android/repository/$ANDROID_TOOLS
+  - unzip -q $ANDROID_TOOLS -d $HOME/sdk
+  # set SDK tools path variable and ANDROID_HOME
+  - export ANDROID_BIN=$HOME/sdk/tools/bin
+  - export ANDROID_HOME=$HOME/sdk
+  - export ANDROID_SDK_ROOT=$HOME/sdk
+  # create empty cfg file to prevent sdkmanager warning message
+  - mkdir -p $HOME/.android && touch $HOME/.android/repositories.cfg
   - chmod +x gradlew
-
-before_cache:
-  - rm -f  $HOME/.gradle/caches/modules-2/modules-2.lock
-  - rm -fr $HOME/.gradle/caches/*/plugin-resolution/
-cache:
-  directories:
-    - $HOME/.gradle/caches/
-    - $HOME/.gradle/wrapper/
-    - $HOME/.android/build-cache
+  - $ANDROID_BIN/sdkmanager --update --sdk_root=$ANDROID_HOME --no_https --verbose
+  - yes | $ANDROID_BIN/sdkmanager --install "tools" --sdk_root=$ANDROID_HOME --no_https --verbose | grep -v = || true
+  - yes | $ANDROID_BIN/sdkmanager --install "platforms;android-29" --sdk_root=$ANDROID_HOME --no_https --verbose | grep -v = || true
+  - yes | $ANDROID_BIN/sdkmanager --install "build-tools;29.0.3" --sdk_root=$ANDROID_HOME --no_https --verbose | grep -v = || true
+  - yes | $ANDROID_BIN/sdkmanager --install "system-images;android-29;default;x86_64" --sdk_root=$ANDROID_HOME --no_https --verbose | grep -v = || true
+  - $ANDROID_BIN/sdkmanager --update --sdk_root=$ANDROID_HOME --no_https --verbose
+    
+    # Set up KVM on linux for hardware acceleration. Manually here so it only happens for emulator tests, takes ~30s
+  - travis_retry sudo -E apt-get -yq --no-install-suggests --no-install-recommends install bridge-utils libpulse0 libvirt-bin qemu-kvm virtinst ubuntu-vm-builder
+  - sudo adduser $USER libvirt
+  - sudo adduser $USER kvm
+
+  #Create emulator
+  - echo no | $ANDROID_BIN/avdmanager create avd --force -n test -k "system-images;android-29;default;x86_64"
+  - sudo $ANDROID_HOME/platform-tools/adb start-server
+  - sudo -E sudo -u $USER -E bash -c "$ANDROID_HOME/emulator/emulator -avd test -no-audio -no-window &"
 
 before_script:
-  - android list targets
-  - echo no | android create avd --force -n test -t android-$EMULATOR_API_VERSION --abi armeabi-v7a
-  - emulator -avd test -no-audio -no-window &
-  - android-wait-for-emulator
-  - adb shell input keyevent 82 &
+  - $ANDROID_HOME/platform-tools/adb wait-for-device
+  - $ANDROID_HOME/platform-tools/adb shell input keyevent 82 & 
```

Comment:
Could these sections of the travis files be loaded from specific place? may be in common? as these seem to be mostly same across repos

Reply:
I'm planning to change the android_complete update versions scripts I wrote to do this... travis expects a file in the specific repo at a specific location so i'm guessing i can't use a specific file.